gdnative-bindings/src/serde.rs

```diff
@@ -0,0 +1,158 @@
+use crate::Object;
+use gdnative_core::core_types::{GodotString, Variant};
+use gdnative_core::object::{Instanciable, RefImplBound, SafeDeref, SubClass};
+use gdnative_core::thread_access::Unique;
+use gdnative_core::{GodotObject, NewRef, Ref};
+use serde::de::{MapAccess, Visitor};
+use serde::ser::SerializeStruct;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::fmt::Formatter;
+use std::marker::PhantomData;
+use std::ops::Deref;
+
+//TODO: Optimize (by adding names to a static HashSet?) instead of leaking so many strings
+
+impl Serialize for Object {
+    #[inline]
+    fn serialize<S>(&self, ser: S) -> Result<<S as Serializer>::Ok, <S as Serializer>::Error>
+    where
+        S: Serializer,
+    {
+        let name = Box::leak(self.get_class().to_string().into_boxed_str());
+        let arr = self.get_property_list();
+        let mut ser = ser.serialize_struct(name, arr.len() as usize)?;
+        for dict in arr.iter().map(|v| v.to_dictionary()) {
+            let name = dict.get("name").to_godot_string();
+            let value = self.get(name.new_ref());
+            let name = Box::leak(name.to_string().into_boxed_str());
+            ser.serialize_field(&*name, &value)?
+        }
+        ser.end()
+    }
+}
+
+/// Newtype that allows serializing all properties of an instance of a Godot class, skipping
+/// properties that are set to their default values. This can be used in conjunction with an
+/// [EditorImportPlugin] or custom engine module to mimic Godot's `tres` files while using a
+/// different syntax.
+pub struct SerializableObject<'a, T: GodotObject>(&'a T);
+
+// Unfortunately, Rust's orphan rules won't allow us to `impl Deserialize` directly for `&T`,
+// and `Object` is defined in this crate, not in `gdnative_core`. Thus we can't add `SubClass<Object>`
+// bounds in `gdnative_core`, and we can only `impl Serialize` on a newtype here.
```

Comment:
Ah, I know I had seen them around but didn't realize how they are used. So if we do include the implementations for `Object`s or `Reference`s, this is probably the better place for it. However, as [Bromeon points out](https://github.com/godot-rust/godot-rust/pull/743#issuecomment-853010497), are we sure it will even be generally useful to serialize these types, or should it be left to custom implementations anyway?